tests/z9999_image_bootstrap_test.py

```diff
@@ -0,0 +1,132 @@
+        image_tag = f'{TAG_PREFIX}{name}'
+
+        # Send output to subprocess.DEVNULL since we EXPECT docker to tell us containers and images don't exist.
+        #   pylint: disable=subprocess-run-check, line-too-long
+        subprocess.run(['docker', 'container', 'rm', container_name, '--force'], stderr=subprocess.DEVNULL, stdout=subprocess.DEVNULL, check=False)
+        subprocess.run(['docker', 'image', 'rm', image_tag], stderr=subprocess.DEVNULL, stdout=subprocess.DEVNULL, check=False)
+
+
+def sub_versions(name, cdt=None, leap=None, refcon=None):
+    """Tests that versions are set."""
+
+    # Sanity check and cleanup assurance.
+    assert name is not None
+    assert name in CLEANUP_NAMES, f"TEST DEFECT: name ({name}) is missing from CLEAN_NAMES array."
+
+    # Set up some constants.
+    container_name = f'{CONTAINER_NAME_PREFIX}{name}'
+    image_tag = f'{TAG_PREFIX}{name}'
+
+    bootstrap_command = [sys.executable, os.path.join(DUNES_ROOT, "bootstrap.py"), f'--tag={image_tag}']
+
+    # Add versions:
+    if cdt:
+        bootstrap_command.append(f'--cdt={cdt}')
+    if leap:
+        bootstrap_command.append(f'--leap={leap}')
+    if refcon:
+        bootstrap_command.append(f'--refcon={refcon}')
+
+    # Execute the bootstrap function.
+    subprocess.run(bootstrap_command, check=True)
+
+    # Start the container
+    subprocess.run(['docker', 'create', '-it', '--name', container_name, image_tag, '/bin/bash'], check=True)
+    subprocess.run(['docker', 'start', container_name], check=True)
+
+    if cdt:
+        # Try to get CDT version info from inside the container and test it matches.
+        #  pylint: disable=line-too-long
+        completed_process = subprocess.run(['docker', 'exec', '-i', container_name, '/usr/bin/ls', '/usr/opt/cdt'], check=False, stdout=subprocess.PIPE)
+        assert cdt in completed_process.stdout.decode()
+
+
+    if leap:
+        # Try to get LEAP version info from inside the container and test it matches.
+        #  pylint: disable=line-too-long
+        completed_process = subprocess.run(['docker', 'exec', '-i', container_name, 'leap-util', 'version', 'client'], check=False, stdout=subprocess.PIPE)
+        assert leap in completed_process.stdout.decode()
+
+    if refcon:
+        # Try to get refcon version info from inside the container and test it matches.
+        #  pylint: disable=line-too-long
+        completed_process = subprocess.run(['docker', 'exec', '-i', container_name, 'leap-util', 'version', 'client'], check=False, stdout=subprocess.PIPE)
+        assert refcon in completed_process.stdout.decode()
+
+
+def test_simple_version():
+    """Ensure bootstrap can create an image with given CDT and LEAP versions."""
+
+    # Start in a clean state.
+    cleanup()
+
+    # Call the test with specific cdt and leap versions.
+    sub_versions('dunes_for_ci', cdt='3.0.1', leap='3.2.1')
+
+    # End in a clean state.
+    cleanup()
+
+
+def test_cdt_leap_versions_non_ci():
```

Comment:
Why we don't want some tests to be executed in CI? Is it because they consume too much time?
If yes I would enable all tests in this PR and create a separate issue to create i.e. nightly build that runs the longest tests.

I am afraid that tests which are not executed in CI will eventually stop working and we won't notice it.

Reply:
I believe the FULL test is excessive and it's absolutely a resource hog. I don't think we want to casually consume our resources on this testing. The basic tests are, I think, adequate to capture most regressions.

I think it's a good idea to periodically run the expensive tests ans so I will write an issue (#211) to enable address that.